pkg/driver/controller.go

```diff
@@ -196,16 +221,50 @@ func (d *Driver) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest)
 		return nil, status.Errorf(codes.Internal, "Failed to create Access point in File System %v : %v", accessPointsOptions.FileSystemId, err)
 	}
 
+	volContext := map[string]string{}
+
+	// Fetch mount target Ip for cross-account mount
+	if roleArn != "" {
+		mountTarget, err := localCloud.DescribeMountTargets(ctx, accessPointsOptions.FileSystemId, azName)
+		if err != nil {
+			klog.Warningf("Failed to describe mount targets for file system %v. Skip using `mounttargetip` mount option: %v", accessPointsOptions.FileSystemId, err)
+		} else {
+			volContext[MountTargetIp] = mountTarget.IPAddress
```

Comment:
What if we don't set this mounttarget ip address and just rely on efs-utils to find the mounttarget? Will that fail?

Reply:
Yeah that would fail since mount happens on `node-deamonset` which is a separate process. 

Since you cannot setup config/credential files on the driver pod, driver pod will not know how to switch credentials for different AWS accounts and will utilize the credentials from default instance role or service account.

Reply:
if roleArn is empty, AzName is ignored. should AzName be respected for cross-AZ mount? Trying to make sense of all scenarios' behavior and make sure we don't break the current behavior. If neither roleArn nor az are provided, then efs-utils should find the mount target.

cross-AZ + cross-Account (both az and roleArn have been specified) 
cross-AZ + same-Account (only az has been specified)
same-AZ + cross-Account (only roleArn has been specified)
same-AZ + same-Account (neither have been specified)

Reply:
If a user wants to use cross-az/same-az for same account, they can make use of `az` mount option, for example they can pass `az=<desired az>` under storage class mount options and efs-utils will figure out the az/mount-target-ip to mount.

However, for cross-account mount, due to the inability to pass credentials to node service / efs-utils, we have to pre-determine the az/mount-target-ip and pass it to efs-utils.

Reply:
hm I didn't even notice az in both mountOptions and parameters tbh. Do you mind putting your comment in a code comment?

I double checked and if both mounttargetip and az set then efs-utils will just ignore az instead of trying to determine.

regarding the inability to pass credentials, there might be a way (in the future) to use the token feature https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/1855-csi-driver-service-account-token so that the node service pod can use the client pod's token to get AWS API creds. but that's another big issue.